### PR TITLE
Rename Master -> Shards in Redis

### PR DESCRIPTION
**Pull request by Neaj-Morshad-101**

### CODE REVIEW

File: apis/kubedb/v1/redis_types.go [hunk 1]
```diff
@@ -151,9 +151,9 @@ const (
 
 type RedisClusterSpec struct {
 	// Number of master nodes. It must be >= 3. If not specified, defaults to 3.
```
Comment (AbdullahAlShaad) [inline apis/kubedb/v1/redis_types.go:153]: Update Comments as well

File: apis/kubedb/v1alpha2/conversion.go [hunk 2]
```diff
@@ -622,10 +622,19 @@ func Convert_v1_MongoDBSpec_To_v1alpha2_MongoDBSpec(in *v1.MongoDBSpec, out *Mon
 	return nil
 }
 
+func Convert_v1alpha2_RedisClusterSpec_To_v1_RedisClusterSpec(in *RedisClusterSpec, out *v1.RedisClusterSpec, s conversion.Scope) error {
+	out.Shards = (*int32)(unsafe.Pointer(in.Master))
+	out.Replicas = (*int32)(unsafe.Pointer(in.Replicas))
```
Comment (AbdullahAlShaad) [inline apis/kubedb/v1alpha2/conversion.go:627]: out.Replicas = in.Replicas + 2

// Replica count should tell how many nodes in the shard including master and replica

File: apis/kubedb/v1alpha2/conversion.go [hunk 3]
```diff
@@ -662,10 +671,19 @@ func Convert_v1alpha2_RedisSpec_To_v1_RedisSpec(in *RedisSpec, out *v1.RedisSpec
 	return nil
 }
 
+func Convert_v1_RedisClusterSpec_To_v1alpha2_RedisClusterSpec(in *v1.RedisClusterSpec, out *RedisClusterSpec, s conversion.Scope) error {
+	out.Master = (*int32)(unsafe.Pointer(in.Shards))
+	out.Replicas = (*int32)(unsafe.Pointer(in.Replicas))
```
Comment (AbdullahAlShaad) [inline apis/kubedb/v1alpha2/conversion.go:676]: // Same change here

**Review by AbdullahAlShaad — CHANGES_REQUESTED** [review]

File: apis/kubedb/v1alpha2/conversion.go (hunk 2 above)
Comment (Neaj-Morshad-101) [reply apis/kubedb/v1alpha2/conversion.go:627]: out.Replicas = in.Replicas + 1 ?

Comment (Neaj-Morshad-101) [reply apis/kubedb/v1alpha2/conversion.go:627]: @AbdullahAlShaad

Comment (AbdullahAlShaad) [reply apis/kubedb/v1alpha2/conversion.go:627]: Yes

File: apis/kubedb/v1alpha2/conversion.go [hunk 4]
```diff
@@ -662,10 +674,22 @@ func Convert_v1alpha2_RedisSpec_To_v1_RedisSpec(in *RedisSpec, out *v1.RedisSpec
 	return nil
 }
 
+func Convert_v1_RedisClusterSpec_To_v1alpha2_RedisClusterSpec(in *v1.RedisClusterSpec, out *RedisClusterSpec, s conversion.Scope) error {
+	out.Master = (*int32)(unsafe.Pointer(in.Shards))
```
Comment (Neaj-Morshad-101) [inline apis/kubedb/v1alpha2/conversion.go:678]: .

File: apis/kubedb/v1alpha2/conversion.go [hunk 5]
```diff
@@ -662,10 +685,33 @@ func Convert_v1alpha2_RedisSpec_To_v1_RedisSpec(in *RedisSpec, out *v1.RedisSpec
 	return nil
 }
 
+func Convert_v1_RedisClusterSpec_To_v1alpha2_RedisClusterSpec(in *v1.RedisClusterSpec, out *RedisClusterSpec, s conversion.Scope) error {
+	if in.Shards != nil {
+		out.Master = new(int32)
```
Comment (Neaj-Morshad-101) [inline apis/kubedb/v1alpha2/conversion.go:690]: panic

**Review by ArnobKumarSaha — APPROVED** [review]

**Pull request merged**